content/articles/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/index.md

```diff
@@ -0,0 +1,276 @@
+
+Let's now load this dataset into our Google Colab notebook.
+
+#### Loading exploratory data analysis packages
+We import two exploratory data analysis packages, pandas and Numpy. We will use pandas to read our CSV file and load it into our Google Colab notebook. 
+Numpy is used to perform mathematical computations in our dataset. It also works with arrays.
+
+```python
+import pandas as pd
+import numpy as np
+```
+Let's use `pandas` to load our dataset.
+
+### Loading dataset
+Use the following command to load the dataset.
+
+```python
+df = pd.read_csv("emotion-dataset.csv")
+```
+To see how the dataset is structured use this command.
+
+```python
+df.head()
+```
+The output is shown below.
+
+![Dataset structure](/engineering-education/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/dataset-structure.png)
+
+From the image above we can see that our dataset has two columns: `Emotion` and `Text`. The emotion column represents the various emotion labels. The text shows all the texts in our dataset.
+
+Let's now show the value count for each emotion. This will give the total number of texts for each emotion label.
+
+```python
+df['Emotion'].value_counts()
+```
+The output is shown below.
+
+![Value count](/engineering-education/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/emotion-value-count.png)
+
+Let's now start cleaning our dataset using Neattext.
+
+### Getting started with Neattext
+As mentioned earlier, Neattext is a Python library that is used to pre-process our dataset. Neattext will clean the text dataset by removing stop words and other noise.
+
+To install Neattext, run this command.
+
+```python
+!pip install neattext
+```
+Let's import `neattext` as follows.
+
+```python
+import neattext.functions as nfx
+```
+To use `neattext`, let's list all the methods and attributes used by `neattext` for data cleaning.
+
+```python
+dir(nfx)
+```
+The output for the methods and attributes is shown below.
+
+![Methods and attributes](/engineering-education/how-to-build-an-nlp-based-emotion-detection-model-using-neattext-and-scikit-learn/methods-attributes.png)
+
+We are interested in only two methods from the list: `remove_stopwords` and `remove_userhandles`.
+
+#### Removing user handles
+The dataset contains some Twitter handles of different users. This is noise to our dataset, and it's unnecessary. We use`remove_userhandles` to remove them from our dataset.
+
+```python
+df['Clean_Text'] = df['Text'].apply(nfx.remove_userhandles)
+```
+We use `apply()` to add `remove_userhandles`. We save the cleaned dataset into a new column named `Clean_Text`.
+
+#### Removing stopwords
+Stopwords is a list of all the commonly used words in any language. The words being common they have little classification power. These words may lead to model bias during training.
```

Comment:
What does this sentence, "The words being common they have little classification power." mean?